### PR TITLE
Reorganizing the quest system

### LINKED ISSUE #344: Recipies shouldn't be unlocked on game start
**Is your feature request related to a problem? Please describe.**
Currently, all recipies are unlocked upon the start of the game. This might be a lot to take in for a new player, and therefore we should limit it.

**Describe the solution you'd like**
With the upcoming story update, we could tie the aquisition of new recipies to the quest system or similar. This will make the game story much more rewarding to play through.

**Additional context**
Partly requires #203 to function fully, but can function without.

### PR DESCRIPTION
**Pull request by BenCheung0422**
Reorganizing the quest system.
Resolves #344
Fixes rendering positioning problem of color code styled texts.
Separates tutorials from the entire quest system.
Recipes are unlocked only when one of the materials are obtained.

### CODE REVIEW

File: src/main/java/minicraft/item/Inventory.java [hunk 1]
```diff
@@ -106,32 +107,47 @@ public int add(int slot, Item item) {
...
 			}
 		} else {
-			items.add(slot, item);
+			addItemToList(slot, item);
 			return 1;
 		}
 	}
 
+	private void addItemToList(Item item) { addItemToList(items.size(), item); }
+	private void addItemToList(int index, Item item) {
```
Comment (Makkkkus) [inline src/main/java/minicraft/item/Inventory.java:137]: Why do you need index here?

Comment (Makkkkus) [inline src/main/java/minicraft/item/Inventory.java:137]: These functions should be renamed so they explain _exactly_ what they are doing. This name does not tell us that it checks for quests.

Good that it is private.

**Review by Makkkkus — COMMENTED** [review]
This is good. Haven't tested it so I can't approve it, but it looks good.

**Comment by BenCheung0422** [comment]
Cool, but it is not ready and enough.

**Comment by BenCheung0422** [comment]
I found that there might be a potential problem that the item handling on both breaking and interacting is not standardized. They are separated everywhere in almost every tiles.

**Comment by BenCheung0422** [comment]
This pull request is now ready.

**Review by ChristofferHolmesland — APPROVED** [review]

**Review by Litorom — DISMISSED** [review]
Skimmed through code looks good. Can’t fully personally test it at this time, however concept and code is good

**Review by ChristofferHolmesland — APPROVED** [review]

**Pull request merged**